index.js

```diff
@@ -1,5 +1,111 @@
 const battleship = () => {
-  return 'The winner is...?'
+
+  //step 1
+
+  //objects of players(works)
+  let player1 = {
+    name: prompt("Player 1, enter your name").toUpperCase(),
+    shipCount: 4,
+    gameBoard:[
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ]]
+    }
+
+    let player2 = {
+    name: prompt("Player 2, enter your name").toUpperCase(),
+    shipCount: 4,
+    gameBoard:[
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ]]
+
+  }
+
+
+  //step 2(notice number of placed ships are not always consistent when ran. sometimes it shows 4 other times 3 maybe a two. not sure why. if you see what's wrong, please give me feedback. thanks!
+
+  //setup of ships
+
+    let ships;
+    let xPos;
+    let yPos;
+      for (i=0; i<player1.shipCount; i++){
+      xPos = Math.floor(Math.random() * 4);
+      yPos = Math.floor(Math.random() * 4);
+        if(parseInt(player1.gameBoard[xPos][yPos])!==1){
+        player1.gameBoard[xPos][yPos] = 1;
+        ships++;
+        }
+      }
+
+      for (i=0; i<player2.shipCount; i++){
+      xPos = Math.floor(Math.random() * 4);
+      yPos = Math.floor(Math.random() * 4);
+        if(parseInt(player2.gameBoard[xPos][yPos])!==1){
+          player2.gameBoard[xPos][yPos] = 1;
+        ships++;
+        }
+      }
+
+// i feel logically that if player1 gameboard is shown then player2 should enter those coordinates to win the game not the other way arround. am i right? if so not sure how to fix.
+  console.log(player1.gameBoard.toString())
+
+
+  //step 3
+  //play game
+
+  //function playGame(){
+  let currentPlayer;
+  let otherPlayer;
+  let count=0;
+    while(true){
+    if(+(count%2)===0){
```

Comment:
This is an interesting condition, can you explain what this is doing?

Reply:
So i was working with another student on this. what i was trying to accomplish is the switching between the two players. I know the boolean true may not be ideal but i didnt know what else to put there. it is suppose to give a remainder or as i understand it half of 2 to go from one player to the next. so if currentPlayer =player1 then continue if not go to player2.